keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -162,6 +168,77 @@ class SqlVerificationRepository(
       }
     }
 
+  /**
+   * Query the repository for the states of multiple contexts.
+   *
+   * This call is semantically equivalent to
+   *    contexts.map { context -> this.getStates(context) }
+   *
+   * However, it's implemented as a single query for efficiency.
+   *
+   * @param contexts a list of verification contexts to query for state
+   * @return a list of maps of verification ids to states, in the same order as the contexts. If there are no
+   *         verification states associated with a context, the resulting map will be empty.
+   */
+  override fun getStatesBatch(contexts: List<VerificationContext>) : List<Map<String, VerificationState>> {
```

Comment:
I was a bit confused by this return type. Can a `VerificationContext` have more than one verification ID? In other words, could it just be either a map of IDs to states, or a list in the same order as the contexts, instead of both?

Reply:
One verification id can apply to many artifact versions

Reply:
Ah. Got it.